README.md
<h1 align="center">Welcome to ts-foursquare 👋</h1>

<p>A simple library for Foursquare API</p>

<p>
  <a href="https://www.npmjs.com/package/ts-foursquare" target="_blank">
    <img alt="Version" src="https://img.shields.io/npm/v/ts-foursquare.svg">
  </a>
  <img alt="Commit Activity" src="https://img.shields.io/github/commit-activity/m/arnaud-zg/ts-foursquare" />
  <a href="https://travis-ci.org/arnaud-zg/ts-foursquare" target="_blank">
    <img alt="Build Status" src="https://travis-ci.org/arnaud-zg/ts-foursquare.svg?branch=develop" />
  </a>
  <a href="https://bundlephobia.com/result?p=ts-foursquare" target="_blank">
    <img alt="Bundle Size" src="https://badgen.net/bundlephobia/min/ts-foursquare" />
  </a>
</p>

<p>
  <img alt="Issues Closed" src="https://img.shields.io/github/issues-closed/arnaud-zg/ts-foursquare">
  <img alt="Issues Open" src="https://img.shields.io/github/issues/arnaud-zg/ts-foursquare">
  <img alt="Pull Request Closed" src="https://img.shields.io/github/issues-pr-closed/arnaud-zg/ts-foursquare">
  <img alt="Pull Request Open" src="https://img.shields.io/github/issues-pr/arnaud-zg/ts-foursquare">
</p>

### Installing

Using npm:

```shell
npm i --save ts-foursquare
```

Using yarn:

```shell
yarn add --dev ts-foursquare
```

## Documentation

Look at the documentation

- [Usage](./docs/usage.md)

## Running the tests

Tests are written with jest

### Unit tests

Using jest:

```shell
yarn run test
```

## Built With

- [TSDX](https://github.com/palmerhq/tsdx) - TSDX

## Contributing

Please read [CONTRIBUTING.md](https://gist.github.com/PurpleBooth/b24679402957c63ec426) for details on our code of conduct, and the process for submitting pull requests to us.

## Versioning

We use [SemVer](http://semver.org/) for versioning. For the versions available, see the [tags on this repository](https://github.com/arnaud-zg/ts-foursquare/tags).

## Authors

- **Arnaud Zheng** - _Initial work_ - [arnaud-zg](https://github.com/arnaud-zg)

See also the list of [contributors](https://github.com/arnaud-zg/ts-foursquare/graphs/contributors) who participated in this project.

## Show your support

Give a ⭐️ if this project helped you!

## License

This project is licensed under the MIT License - see the [LICENSE.md](LICENSE.md) file for details


CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

### [1.0.12](https://github.com/arnaud-zg/ts-foursquare/compare/v1.0.11...v1.0.12) (2020-08-01)


### 🔧 Chore

* **deps:** update dependency @types/jest to v26.0.7 ([#500](https://github.com/arnaud-zg/ts-foursquare/issues/500)) ([aa5a923](https://github.com/arnaud-zg/ts-foursquare/commit/aa5a923199ea53d81ad10850d977a37ffd3a1358))
* **deps:** update dependency @types/jest to v26.0.8 ([#506](https://github.com/arnaud-zg/ts-foursquare/issues/506)) ([ae1be0c](https://github.com/arnaud-zg/ts-foursquare/commit/ae1be0cfff41c27cc6af9e9bfaccc864e69de240))
* **deps:** update dependency @types/node to v12.12.48 ([#490](https://github.com/arnaud-zg/ts-foursquare/issues/490)) ([dccdfd7](https://github.com/arnaud-zg/ts-foursquare/commit/dccdfd7029d19d3443a506f161c94494460cc87d))
* **deps:** update dependency @types/node to v12.12.53 ([#492](https://github.com/arnaud-zg/ts-foursquare/issues/492)) ([637579a](https://github.com/arnaud-zg/ts-foursquare/commit/637579ad8594c35ecda1e68d5537936ac0b4089b))
* **deps:** update dependency codecov to v3.7.2 ([#504](https://github.com/arnaud-zg/ts-foursquare/issues/504)) ([3aea2d5](https://github.com/arnaud-zg/ts-foursquare/commit/3aea2d5fe735b7b18121b69d97b55e55d3582104))
* **deps:** update dependency standard-version to v8.0.2 ([#496](https://github.com/arnaud-zg/ts-foursquare/issues/496)) ([a6e73af](https://github.com/arnaud-zg/ts-foursquare/commit/a6e73af32629bcad673ba78d28f2b23101f0bd79))
* **deps:** update dependency typescript to v3.9.7 ([#497](https://github.com/arnaud-zg/ts-foursquare/issues/497)) ([e4975cd](https://github.com/arnaud-zg/ts-foursquare/commit/e4975cd9d0da357b7d27c6f500098627a832ccb1))
* **deps:** update node.js to v10.22.0 ([#503](https://github.com/arnaud-zg/ts-foursquare/issues/503)) ([6818f84](https://github.com/arnaud-zg/ts-foursquare/commit/6818f84d1a11cdc8a315f32d3a6503fafc1190de))


### 🐛 Bug Fixes

* **deps:** update dependency rxjs to v6.6.2 ([#505](https://github.com/arnaud-zg/ts-foursquare/issues/505)) ([6a9b9ec](https://github.com/arnaud-zg/ts-foursquare/commit/6a9b9ec1c0bf32a248bcd6d7b84fec244910b894))


### 📦 Build System

* **deps:** bump lodash from 4.17.15 to 4.17.19 ([#498](https://github.com/arnaud-zg/ts-foursquare/issues/498)) ([b9047f1](https://github.com/arnaud-zg/ts-foursquare/commit/b9047f121c54fc397fd8ab39e65759db3837cf7d))
* **deps-dev:** bump codecov from 3.7.0 to 3.7.1 ([#502](https://github.com/arnaud-zg/ts-foursquare/issues/502)) ([22b6d99](https://github.com/arnaud-zg/ts-foursquare/commit/22b6d994103257235f937452579fb08e2369693b))
* **deps-dev:** bump standard-version from 8.0.0 to 8.0.1 ([#494](https://github.com/arnaud-zg/ts-foursquare/issues/494)) ([d457e8a](https://github.com/arnaud-zg/ts-foursquare/commit/d457e8a10b1555180237d2a5b1f39323228acdfc))

### [1.0.11](https://github.com/arnaud-zg/ts-foursquare/compare/v1.0.10...v1.0.11) (2020-07-07)


### 🔧 Chore

* **deps:** update dependency @types/jest to v26.0.3 ([#486](https://github.com/arnaud-zg/ts-foursquare/issues/486)) ([5bd4903](https://github.com/arnaud-zg/ts-foursquare/commit/5bd49033a42a02c83e0aa73296ba2b7307423047))
* **deps:** update dependency @types/jest to v26.0.4 ([#489](https://github.com/arnaud-zg/ts-foursquare/issues/489)) ([4892fa9](https://github.com/arnaud-zg/ts-foursquare/commit/4892fa9ef5cdae9eb5c4725039a8a73045c25923))
* **deps:** update dependency typescript to v3.9.6 ([#487](https://github.com/arnaud-zg/ts-foursquare/issues/487)) ([117b974](https://github.com/arnaud-zg/ts-foursquare/commit/117b974d9d66df448599b351e395a1b6b5b6f682))


### 🐛 Bug Fixes

* update renovate.json ([22c2dc3](https://github.com/arnaud-zg/ts-foursquare/commit/22c2dc3838276a07da94b3d6f69570b84f9b77b0))
* **deps:** update dependency http-response-status to v1.0.17 ([#491](https://github.com/arnaud-zg/ts-foursquare/issues/491)) ([c256318](https://github.com/arnaud-zg/ts-foursquare/commit/c256318eb4147d85e87efbdfae6ff26b1c4af327))
* **deps:** update dependency rxjs to v6.6.0 ([#488](https://github.com/arnaud-zg/ts-foursquare/issues/488)) ([ee09ed1](https://github.com/arnaud-zg/ts-foursquare/commit/ee09ed189416b14821d8a868a0ff346b1f829a29))

### [1.0.10](https://github.com/arnaud-zg/ts-foursquare/compare/v1.0.9...v1.0.10) (2020-06-18)


### 🐛 Bug Fixes

* update renovate.json ([a76e4df](https://github.com/arnaud-zg/ts-foursquare/commit/a76e4dfb943b76aa691b9093d359db3be610136e))
* **deps:** update dependency http-response-status to v1.0.16 ([#484](https://github.com/arnaud-zg/ts-foursquare/issues/484)) ([f548c2a](https://github.com/arnaud-zg/ts-foursquare/commit/f548c2a4e42741f54b1de014a4551c2aecc0f275))

### [1.0.9](https://github.com/arnaud-zg/ts-foursquare/compare/v1.0.8...v1.0.9) (2020-06-11)


### 🔧 Chore

* **deps:** update dependency @types/jest to v26 ([#483](https://github.com/arnaud-zg/ts-foursquare/issues/483)) ([71472d6](https://github.com/arnaud-zg/ts-foursquare/commit/71472d6bc85d637d397048cb9dbc6f190cb40e8a))
* **deps:** update dependency @types/node to v12.12.47 ([#480](https://github.com/arnaud-zg/ts-foursquare/issues/480)) ([30fd73b](https://github.com/arnaud-zg/ts-foursquare/commit/30fd73b6800c53fe13d3675b390ce97642c1a966))
* **deps:** update dependency typescript to v3.9.5 ([#482](https://github.com/arnaud-zg/ts-foursquare/issues/482)) ([bc820d8](https://github.com/arnaud-zg/ts-foursquare/commit/bc820d827dd02b2817631217ff6e56cb09c03494))
* **deps:** update node.js to v10.21.0 ([#481](https://github.com/arnaud-zg/ts-foursquare/issues/481)) ([522016e](https://github.com/arnaud-zg/ts-foursquare/commit/522016e05b54a7e70a6f3eb6b89b17a89530e5f1))


### 🐛 Bug Fixes


types/http.ts
export namespace NHttpHeaders {
  export enum EHttpHeaders {
    ACCEPT = 'Accept',
    CONTENT_TYPE = 'Content-Type',
    APPLICATION_JSON = 'application/json',
  }
}

export namespace NHttpStatuses {
  export enum EInformational {
    CONTINUE = 100,
    SWITCHING_PROTOCOLS = 101,
    PROCESSING = 102,
  }

  export enum ESuccess {
    OK = 200,
    CREATED = 201,
    ACCEPTED = 202,
    NON_AUTHORITATIVE_INFORMATION = 203,
    NO_CONTENT = 204,
    RESET_CONTENT = 205,
    PARTIAL_CONTENT = 206,
    MULTI_STATUS = 207,
    ALREADY_REPORTED = 208,
    IM_USED = 226,
  }

  export enum ERedirection {
    MULTIPLE_CHOICES = 300,
    MOVED_PERMANENTLY = 301,
    FOUND = 302,
    SEE_OTHER = 303,
    NOT_MODIFIED = 304,
    USE_PROXY = 305,
    TEMPORARY_REDIRECT = 307,
    PERMANENT_REDIRECT = 308,
  }

  export enum ClientError {
    BAD_REQUEST = 400,
    UNAUTHORIZED = 401,
    PAYMENT_REQUIRED = 402,
    FORBIDDEN = 403,
    NOT_FOUND = 404,
    METHOD_NOT_ALLOWED = 405,
    NOT_ACCEPTABLE = 406,
    PROXY_AUTHENTICATION_REQUIRED = 407,
    REQUEST_TIMEOUT = 408,
    CONFLICT = 409,
    GONE = 410,
    LENGTH_REQUIRED = 411,
    PRECONDITION_FAILED = 412,
    PAYLOAD_TOO_LARGE = 413,
    REQUEST_URI_TOO_LONG = 414,
    UNSUPPORTED_MEDIA_TYPE = 415,
    REQUESTED_RANGE_NOT_SATISFIABLE = 416,
    EXPECTATION_FAILED = 417,
    I_M_A_TEAPOT = 418,
    MISDIRECTED_REQUEST = 421,
    UNPROCESSABLE_ENTITY = 422,
    LOCKED = 423,
    FAILED_DEPENDENCY = 424,
    UPGRADE_REQUIRED = 426,
    PRECONDITION_REQUIRED = 428,
    TOO_MANY_REQUESTS = 429,
    REQUEST_HEADER_FIELDS_TOO_LARGE = 431,
    CONNECTION_CLOSED_WITHOUT_RESPONSE = 444,
    UNAVAILABLE_FOR_LEGAL_REASONS = 451,
    CLIENT_CLOSED_REQUEST = 499,
  }

  export enum EServerError {
    INTERNAL_SERVER_ERROR = 500,
    NOT_IMPLEMENTED = 501,
    BAD_GATEWAY = 502,
    SERVICE_UNAVAILABLE = 503,
    GATEWAY_TIMEOUT = 504,
    HTTP_VERSION_NOT_SUPPORTED = 505,
    VARIANT_ALSO_NEGOTIATES = 506,
    INSUFFICIENT_STORAGE = 507,
    LOOP_DETECTED = 508,
    NOT_EXTENDED = 510,
    NETWORK_AUTHENTICATION_REQUIRED = 511,
    NETWORK_CONNECT_TIMEOUT_ERROR = 599,
  }
}
